homeassistant/components/husqvarna_automower/coordinator.py

```diff
@@ -66,24 +68,28 @@ async def client_listen(
         hass: HomeAssistant,
         entry: ConfigEntry,
         automower_client: AutomowerSession,
-        reconnect_time: int = 2,
     ) -> None:
         """Listen with the client."""
         try:
             await automower_client.auth.websocket_connect()
-            reconnect_time = 2
             await automower_client.start_listening()
+            # Reset reconnect time after successful connection
+            self.reconnect_time = DEFAULT_RECONNECT_TIME
```

Comment:
We need to reset the reconnect time before the listen call since that will block if it's successful.
```suggestion
            # Reset reconnect time after successful connection
            self.reconnect_time = DEFAULT_RECONNECT_TIME
            await automower_client.start_listening()
```